src/core/thread/mle.cpp

```diff
@@ -4446,5 +4464,27 @@ void Mle::UpdateAllDomainBackboneRouters(BackboneRouter::Leader::DomainPrefixSta
 }
 #endif // OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
 
+// Since Thread 1.2, MTD end device (including rx-on-when-idle or rx-off-when-idle) should
+// register or de-register their multicast address larger than realm local to its Thread 1.2
+// parent via MLE Child Update Request.
+bool Mle::ShouldRegisterMulticastToParent(void) const
+{
+    bool update = false;
```

Comment:
How about this:
(under 1.2 we are practically adding one more check for non-sleepy cases that parent is 1.2 or higher).
```
     return IsChild() && !IsFullThreadDevice() &&
           (!IsRxOnWhenIdle()
#if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
            || !GetParent().IsThreadVersion1_1()
#endif
           ); 
```

Also since we seem to be using `ShouldRegisterMultciast` only in one place, we can consider inlining the same check in the code and remve `ShouldRegister` method. Thoughts?

Reply:
SGTM. Removed.